lib/matplotlib/backend_bases.py

```diff
@@ -3042,7 +3042,10 @@ def release_zoom(self, event):
             x, y = event.x, event.y
             lastx, lasty, a, ind, view = cur_xypress
             # ignore singular clicks - 5 pixels is a threshold
-            if abs(x - lastx) < 5 or abs(y - lasty) < 5:
+            # allows the user to "cancel" a zoom action
+            # by zooming by less than 5 pixels
+            if ((abs(x - lastx) < 5 and self._zoom_mode!="y") or
+                (abs(y - lasty) < 5 and self._zoom_mode!="x")):
```

Comment:
Can you put an extra 4 spaces on this line?  This is one of those cases where I am pretty sure the pep8 guidelines are contradictory.  The over-indent is less bad than aligning with the body.